src/nvim/ex_cmds2.c

```diff
@@ -744,40 +744,38 @@ void dbg_breakpoint(char_u *name, linenr_T lnum)
 
 static char_u   *profile_fname = NULL;
 
-/*
- * ":profile cmd args"
- */
+/// :profile cmd args
 void ex_profile(exarg_T *eap)
 {
   static proftime_T pause_time;
 
-  char_u      *e;
-  int len;
+  char_u *arg = skiptowhite(eap->arg);
+  int len     = (int)(arg - eap->arg);
+  arg         = skipwhite(arg);
 
-  e = skiptowhite(eap->arg);
-  len = (int)(e - eap->arg);
-  e = skipwhite(e);
+  const char *cmd = (char *)eap->arg;
 
-  if (len == 5 && STRNCMP(eap->arg, "start", 5) == 0 && *e != NUL) {
+  if (len == 5 && !strncmp(cmd, "start", 5) && *arg) {
     free(profile_fname);
-    profile_fname = vim_strsave(e);
+    profile_fname = expand_env_save_opt(arg, true);
     do_profiling = PROF_YES;
     profile_set_wait(profile_zero());
     set_vim_var_nr(VV_PROFILING, 1L);
-  } else if (do_profiling == PROF_NONE)
-    EMSG(_("E750: First use \":profile start {fname}\""));
-  else if (STRCMP(eap->arg, "pause") == 0) {
```

Comment:
I do agree to explicit comparison for `strcmp`. `strequal` would be a good addition, too. Maybe better defined as an inline function instead of a macro.